test/unit/condition_unit_test.rb

```diff
@@ -166,6 +166,37 @@ def test_default_context_is_deprecated
     assert_includes(err.lines.map(&:strip), expected)
   end
 
+  def test_parse_expression_in_strict_mode
+    environment = Environment.build(error_mode: :strict)
+    parse_context = ParseContext.new(environment: environment)
+    result = Condition.parse_expression(parse_context, 'product.title')
+
+    assert_instance_of(VariableLookup, result)
+    assert_equal('product', result.name)
+    assert_equal(['title'], result.lookups)
+  end
+
+  def test_parse_expression_in_rigid_mode_raises_internal_error
```

Comment:
Not 100% sold on the value of these tests since `Condition` doesn't actually implement this behavior itself. I'm not opposed to leaving it in but not sure if it's actually testing something.